src/html5sortable.ts

```diff
@@ -577,185 +580,225 @@ export default function sortable (sortableElements, options) {
+      dragging.oldDisplay = dragging.style.display
+    }
 
-      dragging.style.display = dragging.oldDisplay
-      delete dragging.oldDisplay
+    if (dragging.style.display !== 'none') {
+      dragging.style.display = 'none'
+    }
 
+    if (placeholderIndex < thisIndex) {
+      _after(element, placeholder)
+    } else {
+      _before(element, placeholder)
+    }
+    // Intentionally violated chaining, it is more complex otherwise
+    placeholders
+      .filter(function (element) { return element !== placeholder })
+      .forEach(_detach)
+  } else {
+    if (placeholders.indexOf(element) === -1 &&
+        !_filter(_getChildren(element), options.items).length) {
       placeholders.forEach(_detach)
-      newParent = this.parentElement
-
-      if (_listsConnected(newParent, startParent)) {
-        sortableElement.dispatchEvent(_makeEvent('sortstop', {
-          item: dragging,
-          startparent: startParent
-        }))
-        if (index !== _index(dragging) || startParent !== newParent) {
-          sortableElement.dispatchEvent(_makeEvent('sortupdate', {
-            item: dragging,
-            index: _filter(_getChildren(newParent), _data(newParent, 'items'))
-              .indexOf(dragging),
-            oldindex: items.indexOf(dragging),
-            elementIndex: _index(dragging),
-            oldElementIndex: index,
-            startparent: startParent,
-            endparent: newParent,
-            newEndList: _serialize(newParent),
-            newStartList: _serialize(startParent),
-            oldStartList: startList
-          }))
-        }
-      }
-      dragging = null
-      draggingHeight = null
-    })
-    // Handle drop event on sortable & placeholder
-    // TODO: REMOVE placeholder?????
-    _on([sortableElement, placeholder], 'drop', function (e) {
-      var visiblePlaceholder
-      if (!_listsConnected(sortableElement, dragging.parentElement)) {
-        return
-      }
-      e.preventDefault()
-      e.stopPropagation()
+      element.appendChild(placeholder)
+    }
+  }
+}
 
-      _data(dragging, 'dropped', 'true')
-      visiblePlaceholder = placeholders.filter(_attached)[0]
-      _after(visiblePlaceholder, dragging)
-    })
+// Handle dragover and dragenter events on draggable items
+var onDragOverEnter = function (e) {
+  var sortableElement = e.target.parentElement
+  if (!dragging || !_listsConnected(sortableElement, dragging.parentElement) || _data(sortableElement, '_disabled') === 'true') {
+    return
+  }
 
-    var debouncedDragOverEnter = _debounce(function (element, pageY) {
-      if (!dragging) {
-        return
-      }
+  var options = _data(sortableElement, 'opts')
+  var maxItems = 0
+  // max items
+  if (options.maxItems && typeof options.maxItems === 'number') {
+    maxItems = options.maxItems
+  }
+  if (maxItems && _filter(_getChildren(sortableElement), _data(sortableElement, 'items')).length >= maxItems) {
+    return
+  }
```

Comment:
I think this could be improved (I know it was like this before).

```js
if (parseInt(options.maxItems) && _filter(_getChildren(sortableElement), _data(sortableElement, 'items')).length >= parseInt(options.maxItems) ) {
   return
}
```

What do you think?

Reply:
Definitely could be improved. I was trying not to change too much code at once.